src/ng/interval.js

```diff
@@ -188,6 +188,8 @@ function $IntervalProvider() {
       */
     interval.cancel = function(promise) {
       if (promise && promise.$$intervalId in intervals) {
+        // Interval cancels should not report as unhandled promise.
+        intervals[promise.$$intervalId].promise.then(noop, noop);
```

Comment:
Isn't this just `promise.then(noop, noop)` ? (Of course the user can mess things up by attaching the `$$intervalId` to an arbitrary promise, but it's (hopefully) unlikely.)

Unrelated to this PR, but any idea why we mark the `promise` parameter as optional ?